src/main/java/service/PriceCatalogHorizontalImpl.java

```diff
@@ -3,16 +3,14 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public class PriceCatalogImpl implements PriceCatalog {
+public class PriceCatalogHorizontalImpl implements PriceCatalogHorizontal {
 
     private final Map<Integer, Double> colorMap = new HashMap<>();
+    private DataReader dataReader;
 
-    @Override
-    public Map<Integer, Double> initDate() {
-        colorMap.put(201, 8.8);
-        colorMap.put(202, 10.4);
-
-        return colorMap;
+    public PriceCatalogHorizontalImpl(DataReader dataReader) {
+        this.dataReader = dataReader;
+        dataReader.readPricesFromFile(colorMap, dataReader.getPropertyValue("horizontalCatalog"));
```

Comment:
It is back practice to call instance method inside a constructor. You can refactor it by implementing a cache

Reply:
Method calls in the constructor are excluded.